.github/actions/component-diff/action.yaml

```diff
@@ -0,0 +1,111 @@
+name: component-diff
+description: |
+  An opinionated action to determine the component-diff between the current version (read from
+  base-component-descriptor) and the greatest published release-version. Because the component-
+  descriptor and OCM releases repository is read from it, the `prepare.yaml` workflow must be
+  run prior to this action.
+
+inputs:
+  ocm-repositories:
+    description: |
+      a (comma-separated) list of ocm-repository-URLs where component-versions are to be looked up
+      (order is honoured).
+
+      Caveat: this action will only honour final versions. Hence, at least one ocm-repository
+      containing final/release versions must be passed.
+
+outputs:
+  component-diff:
+    description: the determined component diff between the current and the greatest version
+    value: ${{ steps.component-diff.outputs.component-diff }}
+
+runs:
+  using: composite
+  steps:
+    - uses: gardener/cc-utils/.github/actions/install-gardener-gha-libs@master
+    - name: collect-component-descriptor
+      id: component-descriptor
+      uses: gardener/cc-utils/.github/actions/merge-ocm-fragments@master
+      with:
+        component-descriptor-artefact: base-component-descriptor
+    - name: Create component diff
+      id: component-diff
+      shell: python
+      run: |
+        import os
+        import sys
+        import textwrap
+
+        import yaml
+
+        import cnudie.retrieve
+        import oci.client
+        import ocm
+        import version
+
+
+        current_component_descriptor = ocm.ComponentDescriptor.from_dict(
+          yaml.safe_load('''${{ steps.component-descriptor.outputs.component-descriptor}}'''),
+        )
+        component_name = current_component_descriptor.component.name
+        component_version = current_component_descriptor.component.version
+
+        ocm_repositories = '${{ inputs.ocm-repositories }}'.split(',')
+        ocm_repository_lookup = cnudie.retrieve.ocm_repository_lookup(*ocm_repositories)
+
+        oci_client = oci.client.Client()
```

Comment:
I think we should rather use an OCI-Client what will honour "docker-auth", to at least prepare _some_ means of authentication, where required.

btw, there is also [this](https://github.com/gardener/cc-utils/pull/1282) related pullrequest, which I forgot to publish some days ago, which might come in handy